charts/argo-cd/templates/_helpers.tpl

```diff
@@ -174,7 +174,10 @@ Merge Argo Configuration with Preset Configuration
 {{- $config := (mergeOverwrite (deepCopy (omit .Values.configs.cm "create" "annotations")) (.Values.server.config | default dict))  -}}
 {{- $preset := include "argo-cd.config.cm.presets" . | fromYaml | default dict -}}
 {{- range $key, $value := mergeOverwrite $preset $config }}
-{{ $key }}: {{ toString $value | toYaml }}
+{{- $fmted := $value | toString }}
+{{- if not (eq $fmted "") }}
```

Comment:
I would consider just simple check as null or empty string evaluate to `False` and type conversion is not really needed. Construct `{{ $key }}: {{ toString $value | toYaml }}` is used only because users tend to pass all sorts of types into key / value pairs and toYaml expects string.

```suggestion
{{- if not $value }}
```